src/fheroes2/editor/editor_spell_selection.cpp

```diff
@@ -138,14 +138,17 @@ namespace
                 return false;
             }
 
-            if ( eventProcessor.MouseClickLeft() ) {
-                assert( static_cast<size_t>( spellIndex ) < _spellRoi.size() );
+            assert( static_cast<size_t>( spellIndex ) < _spellRoi.size() );
 
+            const fheroes2::Rect spellRoi = _spellRoi[spellIndex];
```

Comment:
Hi, @oleg-derevenetz, should we convert this to `const &` not to make a copy?
```suggestion
            const fheroes2::Rect & spellRoi = _spellRoi[spellIndex];
```

Reply:
@Districh-ru we can do it, but I'm not sure that we should. `Rect` is 4 32-bit ints, which rougly correspond to 2 64-bit pointers, it's not a some structure of a few hundreds or thousands bytes in size, so I doubt that we will win significantly here in terms of performance. Also I prefer not to take a references to items of externally-managed vectors without any particular need because they can easily become dangling if reallocation of this vector will be caused somewhere between.

Reply:
@oleg-derevenetz, thanks for the explanation!
I'll take this into account.